cowprotocol/cowswap
# Add global state to wallet (decouple web3-react)

# Summary

This PR adds the state of the wallet as a joitai state. 

This way it the app don't need to depend from any wallet library anymore. It will use `useWalletInfo` and `useWalletDetails` hooks, which just read from the global state.

There's a new updater, which deals with the wallet library itself. 

Here is an example of autoconnecting to Metamask and then connecting to Impersonator using Wallet Connect. Because the updater memorize the values, it should not update the joitai state too often.

<img width="1678" alt="image" src="https://user-images.githubusercontent.com/2352112/222553210-47de1cdc-68bb-40cd-8a76-e889ed541f3f.png">


# To Test

Try to connect using some representing wallets to make sure I didn't break any important use case.
